### PR TITLE
DetailsList,TextField,Button: minor style fixes.

### PR DESCRIPTION
**Pull request by dzearing**
#### Pull request checklist

- [X] Include a change request file if publishing 
- [X] New feature, bugfix, or enhancement

#### Description of changes

DetailsList: header buttons now use background: transparent, and have the right font styling. 

TextField and Button: ms-u-screenReaderOnly styling dependency removed, mixing in the right styles from core to remove the dependency.

### CODE REVIEW

File: apps/fabric-examples/src/AppDefinition.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { App as AppBase, IAppDefinition, ExampleStatus } from '@uifabric/example-app-base';
```
Comment (micahgodbolt) [inline apps/fabric-examples/src/AppDefinition.tsx]: Was having this change in this PR intentional?

Comment (dzearing) [inline apps/fabric-examples/src/AppDefinition.tsx]: @micahgodbolt yeah just did it while i was there. I beileve it's unnecessary and confusing, but didn't really want to split out a separate pr.
